lib/cache/resource.js

```diff
@@ -81,10 +98,13 @@ var put = function(resource, tags) {
 
     putNestedResources.call(this, clonedResource);
 
-    return this._cache.put(clonedResource.$id, clonedResource, clonedResource.$expires, tags);
+    var cacheKey = this.generateCacheKey(clonedResource.$id, query);
+    var tags = [resource.$id];
+
+    return this._cache.put(cacheKey, clonedResource, clonedResource.$expires, tags);
 };
 
-var putNestedResources = function(obj, tags) {
```

Comment:
It wasn't used, at least not here, so I removed it and had the `ResourceCache` use its own strategy for tagging resource caches.